mongo/dataset_store.go

```diff
@@ -0,0 +1,160 @@
+package mongo
+
+import (
+	"context"
+	"errors"
+	"strconv"
+
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+
+	mongolib "github.com/ONSdigital/dp-mongodb"
+
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	mongoHealth "github.com/ONSdigital/dp-mongodb/health"
+	errs "github.com/ONSdigital/dp-observation-api/apierrors"
+	"github.com/ONSdigital/dp-observation-api/models"
+)
+
+// Mongo represents a simplistic MongoDB configuration, and holds the session and healthclient
+type Mongo struct {
+	Collection   string
+	Database     string
+	URI          string
+	session      *mgo.Session
+	healthClient *mongoHealth.CheckMongoClient
+}
+
+const (
+	editionsCollection = "editions"
+)
+
+// Init creates a new mgo.Session for this Mongo object, with a strong consistency and a write mode of "majortiy".
+func (m *Mongo) Init() (s *mgo.Session, err error) {
+	if m.session != nil {
+		return nil, errors.New("session already exists")
+	}
+
+	// Create session
+	if m.session, err = mgo.Dial(m.URI); err != nil {
+		return nil, err
+	}
+	m.session.EnsureSafe(&mgo.Safe{WMode: "majority"})
+	m.session.SetMode(mgo.Strong, true)
+
+	// Create healthclient from client using the created session
+	client := mongoHealth.NewClient(m.session)
+	m.healthClient = &mongoHealth.CheckMongoClient{
+		Client:      *client,
+		Healthcheck: client.Healthcheck,
+	}
+
+	return m.session, nil
+}
+
+// Session is a getter for the MongoDB session
+func (m *Mongo) Session() *mgo.Session {
+	return m.session
+}
+
+// Close closes teh mongoDB session using dp-mongo library
+func (m *Mongo) Close(ctx context.Context) error {
+	return mongolib.Close(ctx, m.session)
+}
+
+// Checker calls the mongo health client Checker
+func (m *Mongo) Checker(ctx context.Context, state *healthcheck.CheckState) error {
+	if m.healthClient == nil {
+		return errors.New("client not initialised")
+	}
+	return m.healthClient.Checker(ctx, state)
+}
+
+// GetDataset retrieves a dataset document
+func (m *Mongo) GetDataset(id string) (*models.DatasetUpdate, error) {
```

Comment:
🤔 I think possibly a team discussion on this might be needed. We are duplicating functionality of the dataset API here which is not optimal. I wonder if we could call the dataset API to use the functionality it already has.